### DISCUSSION TITLE
Switching to Groq Whisper Large V3 for Cost-Effective and Faster Transcriptions

### DISCUSSION THREAD

**Discussion by Teo-Selcuk** [savbell/whisper-writer#57] [Q&A] [answered]
Hello everyone,

I've been using the amazing [Whisper Writer](https://github.com/savbell/whisper-writer) app, which leverages OpenAI's Whisper model for speech-to-text transcription. Recently, I explored using Groq's Whisper Large V3 API instead, due to its significantly lower costs and higher speed.

**Reasons for Switching:**
1. **Cost Efficiency:** Groq Whisper Large V3 costs only $0.03 per hour of transcription, which is up to 12 times cheaper than OpenAI's Whisper, which costs $0.36 per hour.
2. **Speed:** Groq's model operates at approximately 172 times real-time speed, ensuring faster transcription compared to OpenAI's Whisper.

**Issues Encountered:**
Despite successfully changing the base URL and using Groq's API key, I couldn't modify the model to Whisper Large V3. The settings seem fixed to OpenAI's model.

**Reference:**
- [Groq Speech-to-Text Documentation](https://console.groq.com/docs/speech-text)

Has anyone managed to configure Whisper Writer to use Groq's API effectively? Any insights or guidance would be greatly appreciated!

Thanks in advance!

**Comment by CXwudi** [comment]
I managed to do this with the following `config.yaml`:

```yml
model_options:
  api:
    base_url: https://api.groq.com/openai/v1
    model: whisper-large-v3
```

For some reason, my api_key is saved in a .env file instead of inside yaml file, but probably doesn't matter

**Reply by Teo-Selcuk** [reply]
Hello CXwudi,

Thanks for your detailed response and the configuration suggestion! I tried implementing the changes you recommended to use Groq's Whisper Large V3 in Whisper Writer. However, I'm still encountering some issues. Here's the error message I'm getting:

```
File ".../src/result_thread.py", line 67, in run
    result = transcribe(self.config, audio_file, self.local_model)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File ".../src/transcription.py", line 88, in transcribe
    transcription = transcribe_api(config, audio_file)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File ".../src/transcription.py", line 57, in transcribe_api
    response = client.audio.transcriptions.create(model=model_options['api']['model'],
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File ".../openai/resources/audio/transcriptions.py", line 116, in create
    return self._post(
           ^^^^^^^^^^^
File ".../openai/_base_client.py", line 1240, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File ".../openai/_base_client.py", line 921, in request
    return self._request(
           ^^^^^^^^^^^^^^
File ".../openai/_base_client.py", line 1020, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.AuthenticationError: Error code: 401 - {'error': {'message': 'Invalid API Key', 'type': 'invalid_request_error', 'code': 'invalid_api_key'}}
```

It seems like an authentication issue, and I've made sure to use the correct Groq API key. Here are my current `config.yaml` and `.env` file configurations:

**config.yaml:**
```yaml
  api:
    # api_key: null # default was null
    base_url:  https://api.groq.com/openai/v1
    model: whisper-large-v3 # Other groq model distil-whisper-large-v3-en
```
- I commented out the api_key as mine is the `.env` file too 

**.env file:**
```
OPENAI_API_KEY='....sdsd....' # placed Groq API key instead of OpenAI API key
```

**_Note:_**  The config.yaml file is in the src folder and the .env file is in the root directory.

Despite this setup, I'm still not able to get it to work. I also came across an article recently mentioning that Groq has released a new speech model, "distil-whisper-large-v3-en" which is even cheaper than Whisper Large V3. I was wondering if you've had any experience with this model or have any insights on how to set it up in Whisper Writer.

**Reply by CXwudi** [reply]
`openai.AuthenticationError: Error code: 401 - {'error': {'message': 'Invalid API Key', 'type': 'invalid_request_error', 'code': 'invalid_api_key'}}`

Looks like API key is not picked up, maybe try pasting the API key in the GUI? This is how I setup groq too. I actually just did everything on GUI

**Reply by Teo-Selcuk** [reply]
![image](https://github.com/user-attachments/assets/23372e1b-a81b-402e-ba75-43d81fe5dcb9)

I cannot change the model name thats the problem how were you able to achieve that ?

**Reply by CXwudi** [reply, marked as answer]
That is really wired, this is what i saw 
![image](https://github.com/user-attachments/assets/0d8f9eca-7f72-4c66-a187-fdfb260ef9bb)

I simply just git pull the latest code

**Reply by Teo-Selcuk** [reply]
Thank you so much. I finally got it to work

**Reply by tanurus** [reply]
Wow this is amazing! thank you very much
